gapis/api/vulkan/image_primer.go

```diff
@@ -1263,26 +1293,46 @@ func (h *ipRenderHandler) render(job *ipRenderJob, tsk *scratchTask) error {
 		h.sb.write(h.sb.cb.VkDestroyBuffer(dev, indexBuf, memory.Nullptr))
 	})
 
-	inputBarriers := []VkImageMemoryBarrier{}
+	inputSrcBarriers := []VkImageMemoryBarrier{}
+	dstBarriers := []VkImageMemoryBarrier{}
 	for _, input := range job.inputAttachmentImages {
-		inputLevel := input.Aspects().Get(VkImageAspectFlagBits_VK_IMAGE_ASPECT_COLOR_BIT).Layers().Get(job.layer).Levels().Get(job.level)
-		inputBarriers = append(inputBarriers,
+		aspects := imageBarrierAspectFlags(input.aspect, input.image.Info().Fmt())
+		inputSrcBarriers = append(inputSrcBarriers,
 			NewVkImageMemoryBarrier(h.sb.ta,
 				VkStructureType_VK_STRUCTURE_TYPE_IMAGE_MEMORY_BARRIER, // sType
 				0, // pNext
 				VkAccessFlags((VkAccessFlagBits_VK_ACCESS_MEMORY_WRITE_BIT-1)|VkAccessFlagBits_VK_ACCESS_MEMORY_WRITE_BIT), // srcAccessMask
 				VkAccessFlags(VkAccessFlagBits_VK_ACCESS_INPUT_ATTACHMENT_READ_BIT),                                        // dstAccessMask
-				inputLevel.Layout(),                                                                                        // oldLayout
+				input.initialLayout,                                                                                        // oldLayout
 				VkImageLayout_VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL,                                                     // newLayout
 				queueFamilyIgnore,                                                                                          // srcQueueFamilyIndex
 				queueFamilyIgnore,                                                                                          // dstQueueFamilyIndex
-				input.VulkanHandle(),                                                                                       // image
+				input.image.VulkanHandle(),                                                                                 // image
 				NewVkImageSubresourceRange(h.sb.ta, // subresourceRange
-					VkImageAspectFlags(VkImageAspectFlagBits_VK_IMAGE_ASPECT_COLOR_BIT), // aspectMask
-					0, // baseMipLevel
-					input.Info().MipLevels(), // levelCount
+					aspects, // aspectMask
+					0,       // baseMipLevel
+					input.image.Info().MipLevels(), // levelCount
+					0, // baseArrayLayer
+					input.image.Info().ArrayLayers(), // layerCount
+				),
+			))
+		dstBarriers = append(dstBarriers,
```

Comment:
1) Prefer to have a check to see if the `finalLayout` and the current layout is the same. Because for primeByRendering jobs, the layout of input attachments are guaranteed to be SHADER_READ_ONLY_OPTIMAL before the rendering (line 80), so this is not necessary for those cases, also the barriers for the input attachments in the `inputSrcBarriers` above.

2) When `render()` is called from places other than `primeByRendering` for aspects other than ONLY stencil, not used  when priming for COLOR or DEPTH aspects, so that layout transitions for input attachments can be missed.

Reply:
Sorry, the 2) point is not valid, and for the 1) not necessary for the `inputSrcBarriers`, I misunderstood the code here.

Reply:
Done